### PR TITLE
TINY-4603: Renaming link options to be more consistent

### PR DESCRIPTION
**Pull request by HAFRMO**
Description of Changes:
Renamed variables to be consistent with other option names.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add-ons such as plugins and themes are no longer constructed using the `new` operator #TINY-8256
 - A number of APIs that were not proper classes, are no longer constructed using the `new` operator #TINY-8322
 - Moved the `print` plugin's functionality to TinyMCE core #TINY-8314
+- The `default_link_target` has been renamed to `link_default_target` for both `link` and `autolink` plugins #TINY-4603
+- The `rel_list` has been renamed to `link_rel_list` for `link` plugins #TINY-4603
+- The `target_list` has been renamed to `link_target_list` for `link` plugins #TINY-4603
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:28]:
```suggestion
- The `rel_list` has been renamed to `link_rel_list` for the `link` plugin #TINY-4603
- The `target_list` has been renamed to `link_target_list` for the `link` plugin #TINY-4603
```

**Review by lnewson — APPROVED** [review]
Looks good thanks! Don't forget to update the PR to be more descriptive though because remember it'll be used as the commit message.

- [ ] Improve PR title

**Review by ltrouton — APPROVED** [review]

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 1 above)
Comment (ashfordneil) [reply modules/tinymce/CHANGELOG.md:28]:
```suggestion
- The `rel_list` option has been renamed to `link_rel_list` for the `link` plugin #TINY-4603
- The `target_list` option has been renamed to `link_target_list` for the `link` plugin #TINY-4603
```

File: modules/tinymce/CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add-ons such as plugins and themes are no longer constructed using the `new` operator #TINY-8256
 - A number of APIs that were not proper classes, are no longer constructed using the `new` operator #TINY-8322
 - Moved the `print` plugin's functionality to TinyMCE core #TINY-8314
+- The `default_link_target` has been renamed to `link_default_target` for both `link` and `autolink` plugins #TINY-4603
```
Comment (ashfordneil) [inline modules/tinymce/CHANGELOG.md:26]:
```suggestion
- The `default_link_target` option has been renamed to `link_default_target` for both `link` and `autolink` plugins #TINY-4603
```

**Review by ashfordneil — APPROVED** [review]

**Pull request merged**